src/box/lua/config/instance_config.lua

```diff
@@ -2325,6 +2325,13 @@ return schema.new('instance_config', schema.record({
                 type = 'string',
             }),
             value = schema.record({
+                -- The instances that can't be appointed by the
+                -- supervised failover coordinator.
+                non_appointable = schema.array({
```

Comment:
Raw idea: it would look simpler if we call it just `read_only`.

Reply:
Hm, sounds good, though the name could be easily misunderstood since the instance could be RW meanwhile marked as a `read_only`. E.g. right after bootstrapping.

Reply:
Hm. Now I'm in doubt about this decision.

Quote:

> Though, it was problematic in case of all instances are marked as non-appointable. I don't want to forbid such configuration. Since bootstrap leader anyway goes RO after a while I suppose it's ok not to take the appointable status into account when choosing the bootstrap leader.

We don't have to explicitly forbid it. Just let all the instances wait a bootstrap leader. It would be a straightforward behavior.

Let's imagine that this `non_appointable` instance is expressed in config as an instance with `priority` = `-inf` and apply all the related logic.

TBH, I would think twice about naming to avoid `non_foo` options.

1. `read_only`
2. `learners`
3. `readers`

Something like this would sound good for me. `read_only` seems preferable, since it is consistent with other APIs.

Terminology questions are always hard, so I would appreciate if you ask for more opinions here.

Reply:
The `learner` word is likely recognized by others, see #6281.

Also, `follower` seems quite friendly term too.

Maybe it is easier to use `learner`/`follower` in sentences comparing to `read_only`, which interferes with other parameters.

Reply:
In cartridge similar option was called `electable` (https://github.com/tarantool/cartridge/commit/06147dd891b89f6ba90d9366892a0822b3fc995f), but I'm ok with `learner/follower`, I'm against `read_only` because it is kinda confusing with `box.cfg.read_only` and `non_appointable` because I don't like the negative prefix)

Reply:
Thank you for quite detailed discussion and experienced opinion on it. I've changed the option name to `learners`